src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -0,0 +1,147 @@
+package com.android.volley.toolbox;
+
+import android.annotation.SuppressLint;
+import com.android.volley.AsyncCache;
+import com.android.volley.VolleyLog;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.channels.CompletionHandler;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * AsyncCache implementation that uses Java NIO's AsynchronousFileChannel to perform asynchronous
+ * disk reads and writes. This should only be used by devices with an API level of 26 or above.
```

Comment:
I think you can remove this sentence and the `@SuppressLint` annotation and just add a `@RequiresApi(Build.VERSION_CODES.O)` instead. This is preferable because:

-  Instead of suppressing all lint checks, it only disables them for APIs added up through O. If you make an API call to something added in P, it will still fail, which is good as we would need additional checks.
-  It will enforce that the caller checks for O+ before using the class (or that it has a RequiresApi annotation of its own)
-  Its self-documenting - no need to repeat it in the Javadoc.

Reply:
Fixed this